src/test/java/io/roastedroot/sqlite4j/ConnectionTest.java

```diff
@@ -425,4 +427,46 @@ public void overrideURIPragmaValuesWithProperties() throws Exception {
         stat.close();
         conn.close();
     }
+
+    @Test
+    public void correctlyOpenDatabase() throws Exception {
+        // succeed
+        assertThatNoException()
+                .isThrownBy(
+                        () ->
+                                DriverManager.getConnection(String.format("jdbc:sqlite:sample1.db"))
+                                        .close());
+        Files.deleteIfExists(Path.of("sample1.db"));
+    }
+
+    @Test
+    public void failsOpenDatabaseNoCreateInConfig() throws Exception {
+        var config = new SQLiteConfig();
+        config.resetOpenMode(SQLiteOpenMode.CREATE);
+
+        assertThatThrownBy(
+                        () ->
+                                DriverManager.getConnection(
+                                        String.format("jdbc:sqlite:sample2.db"),
+                                        config.toProperties()))
+                .hasMessageContaining("Database file doesn't exists");
+    }
+
+    @Test
+    public void failsOpenDatabaseNoCreateInConnectionString() throws Exception {
+        assertThatThrownBy(
+                        () ->
+                                DriverManager.getConnection(
+                                        String.format("jdbc:sqlite:sample3.db?open_mode=0")))
+                .hasMessageContaining("Database file doesn't exists");
+    }
+
+    @Test
+    public void failsOpenDatabaseNoCreateUnavailablePath() throws Exception {
+        assertThatThrownBy(
+                        () ->
+                                DriverManager.getConnection(
+                                        String.format("jdbc:sqlite:/NON-EXISTENT9123/sample4.db")))
```

Comment:
Will the DB file be created if create mode is used in configuration? Because that is the use case I'm interested in. If yes, then all is fine! 😄 If not, then I'd argue that the behavior is still inconsistent as why should relative files be created but not ones with paths?

Reply:
> Will the DB file be created if create mode is used in configuration? 

Yes, when create mode is enabled it will try to create an empty file at the specified location if it doesn't exists.

> why should relative files be created but not ones with paths?

Now all files are treated equally, there is no more difference in relative vs. absolute paths.

I believe the behavior is now expected and correct(I followed the instructions available in the SQLite source code comments), but feel free to verify and open an issue if something is acting up :+1:

Reply:
Maybe a test could be added to check that an absolute path with missing intermediate directories properly creates the DB file when create mode is active?